Libraries/LibWeb/HTML/HTMLSelectElement.h

```diff
@@ -97,6 +97,8 @@ class HTMLSelectElement final
     void update_selectedness();
 
 private:
+    friend class HTMLOptionElement;
```

Comment:
We try to avoid using `friend` because it gives too much access. Instead, we use a `Badge<Foo>` parameter on a public method so that only `Foo` can call it. See an explanation [here](https://awesomekling.github.io/Serenity-C++-patterns-The-Badge/) (I don't think we have other docs about it?)

`update_inner_text_element()` also gets called by `HTMLSelectElement` itself, so you'd need an extra method to have the badge and then call that.